src/farcasterd/runtime.rs

```diff
@@ -946,6 +951,89 @@ impl Runtime {
                     }
                 }
             }
+            Request::FundingInfo(info) => match info {
+                FundingInfo::Bitcoin(swapid, addr, amount) => {
+                    self.funding_btc.insert(swapid, (addr, amount));
+                }
+                FundingInfo::Monero(swapid, addr, amount) => {
+                    if let Ok(addr) = monero::Address::from_str(&addr) {
+                        let amount = monero::Amount::from_pico(amount);
+                        self.funding_xmr.insert(swapid, (addr, amount));
+                    } else {
+                        error!("Monero address not invalid");
+                    };
+                }
+            },
+
+            Request::NeedsFunding(Coin::Monero) if !self.funding_xmr.is_empty() => {
+                use tokio::runtime::Builder;
+                let rt = Builder::new_multi_thread()
+                    .worker_threads(1)
+                    .enable_all()
+                    .build()
+                    .unwrap();
+                rt.block_on(async {
+                    let wallet_client =
+                        monero_rpc::RpcClient::new("http://localhost:18083".to_string());
+                    let wallet = wallet_client.wallet();
+                    // let _ = wallet
+                    //     .create_wallet("test".to_string(), None, "English".to_string())
+                    //     .await;
+                    wallet
+                        .open_wallet("stagenet".to_string(), None)
+                        .await
+                        .expect("The wallet exists, created the line before");
+
+                    let options = monero_rpc::TransferOptions::default();
+                    for (i, (swapid, (addr, amount))) in self.funding_xmr.drain().enumerate() {
+                        let mut destination = HashMap::new();
+                        destination.insert(addr, amount.as_pico());
+                        let transaction = wallet
+                            .transfer(
+                                destination.clone(),
+                                monero_rpc::TransferPriority::Default,
+                                options.clone(),
+                            )
+                            .await
+                            .unwrap();
+                    }
+                });
+            }
```

Comment:
@TheCharlatan this stuff is more debugging hack (to be moved elsewhere), but I dont manage to open wallet stagenet that exists on the folder passed to monero-wallet-rpc